toolkit/tools/imagegen/configuration/packagerepo.go

```diff
@@ -59,6 +74,17 @@ func (p *PackageRepo) IsValid() (err error) {
 		return
 	}
 
+	// Repos for ISO use only (install = false) cannot support custom GPG keys if GPGCheck is enabled. Warn the user even
+	// if we are installing the repo file, as the repo file will not be usable in the installer.
+	if p.GPGCheck && p.GPGKeys != getDefaultGPGKeys() {
+		if !p.Install {
+			err = fmt.Errorf("invalid value for package repo '%s' [GPGKeys] (%s), custom GPG keys are only supported for repos that are installed", p.Name, p.GPGKeys)
+			return
+		} else {
+			logger.Log.Warnf("Warning: Custom GPG keys are only supported for repos that are installed. The repo file for '%s' will not be usable in the installer.", p.Name)
```

Comment:
I wouldn't expect the user to know what we mean by "installer" and this warning message may just be confusing for them. Could we re-word it to instead say that the installed repos will be available only in the built image, but not for the sake of populating it with packages during the image build? Or at least that's what I understand this warning is about.